tests/framework/e2e/cluster.go

```diff
@@ -654,6 +602,20 @@ func (cfg *EtcdProcessClusterConfig) EtcdServerProcessConfig(tb testing.TB, i in
 	}
 }
 
+func values(cfg embed.Config) map[string]string {
+	fs := flag.NewFlagSet("etcd", flag.ContinueOnError)
+	cfg.AddFlags(fs)
+	values := map[string]string{}
+	fs.VisitAll(func(f *flag.Flag) {
+		var value string
+		if value != "false" && value != "0" {
+			value = f.Value.String()
+		}
+		values[f.Name] = value
```

Comment:
```suggestion
		value = f.Value.String()
		if value != "" && value != "false" && value != "0" {
			values[f.Name] = value
		}
```

Reply:
I think the result is different, but I will double check if there impact on the results.

Reply:
ok, makes sense.

Reply:
Needed to revert. The code change makes a difference for flags with default value true, that is set to false during the test.

Reply:
If I have time I will add a unit test for this.

Reply:
You define a variable at line 610 `var value string`, its default value is always ""; then you check its value at 611 `if value != "false" && value != "0"  { ... }`, the if condition will always be true. What's the point of the 2 lines of code?

Reply:
Don't have time to deep dive here, the difference is one code works. Other code fails the tests. I will add unit test to show the difference.

Reply:
And I responded, that after I applied you change the test started to consistently fail and it needs further investigation.

Reply:
With https://github.com/etcd-io/etcd/pull/16747 I have confirmed that your change breaks logic:
```
--- FAIL: TestEtcdServerProcessConfig (0.00s)
    --- FAIL: TestEtcdServerProcessConfig/StrictReconfigCheck (0.00s)
        cluster_test.go:214: diff:   []string{
                ... // 11 identical elements
                "/tmp/fake/member-0",
                "--initial-cluster-token=new",
            +   "--strict-reconfig-check=false",
              }
FAIL
FAIL    go.etcd.io/etcd/tests/v3/framework/e2e  0.051s
FAIL
```

Reply:
Please forget my original suggested change. Instead, change

```
		var value string
		if value != "false" && value != "0" {
			value = f.Value.String()
		}
```
 to

```
value := f.Value.String()
```

Reply:
Changed updated it. PTAL. Now with unit test it should be clear what was the problem.